ProfilerGUI/Source/Configurator/TargetAppModel.cs

```diff
@@ -120,21 +121,51 @@ private void UpdateExecutableMachineType()
                 return;
             }
 
-            MachineTypeUtils.MachineType typeOfExecutable = MachineTypeUtils.GetExecutableType(ApplicationPath);
-            if (typeOfExecutable == MachineTypeUtils.MachineType.Unknown)
-            {
-                logger.Warn("Could not determine if target app is 32 or 64 bit, please set manually");
-                return;
-            }
-
-            if (typeOfExecutable == MachineTypeUtils.MachineType.I386)
-            {
-                ApplicationType = EApplicationType.Type32Bit;
-            }
-            else
-            {
-                ApplicationType = EApplicationType.Type64Bit;
+            if (IsDotNETCoreApplication(ApplicationPath))

```

Comment:
I implemented this separately from `MachineTypeUtils.GetExecutableType`, because .NET Core isn't really a machine type. We could also merge these two heuristics to directly determine the application type. I have no clear preference for either option.

Reply:
Sounds legit :+1: